homeassistant/components/solarlog/sensor.py

```diff
@@ -41,14 +40,5 @@ def __init__(
     @property
     def native_value(self):
         """Return the native sensor value."""
-        if self.entity_description.key == "time":
-            state = as_local(
-                getattr(self.coordinator.data, self.entity_description.key)
-            )
-        else:
-            result = getattr(self.coordinator.data, self.entity_description.key)
-            if self.entity_description.factor:
-                state = round(result * self.entity_description.factor, 3)
-            else:
-                state = result
-        return state
+        state = getattr(self.coordinator.data, self.entity_description.key)
+        return self.entity_description.value(state)
```

Comment:
```suggestion
        raw_attr = getattr(self.coordinator.data, self.entity_description.key)
        if self.entity_description.value:
            return self.entity_description.value(raw_attr)
        return raw_attr
```

Reply:
Thanks, that makes sense and fixes my problem.